client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -234,32 +298,43 @@ function mediaButton( editor ) {
 				return;
 			}
 
-			// The `img` object can be a string or a DOM node. We need a
-			// normalized string to replace the post content markup
-			let imgString;
-			if ( img.outerHTML ) {
-				imgString = img.outerHTML;
+			let mediaString = '';
+
+			if ( mediaHasCaption ) {
+				if ( captionNode.outerHTML ) {
+					mediaString = captionNode.outerHTML;
+
+					captionNode.outerHTML = event.content;
+				} else {
+					mediaString = captionNode;
```

Comment:
I did test this behavior on prod with loading images. (We do handle this properly in the html mode). We can leave this for another PR but I think we'd want this fixed quickly